tests/test_config_parsing/test_parser_error_collection.py

```diff
@@ -0,0 +1,280 @@
+    contents: str
+    is_executable: bool = False
+
+
+@dataclass
+class ExpectedErrorInfo:
+    filename: str = "test.ert"
+    line: Optional[int] = None
+    column: Optional[int] = None
+    end_column: Optional[int] = None
+    other_files: Optional[Dict[str, FileDetail]] = None
+    match: Optional[str] = None
+
+
+def write_files(files: Optional[Dict[str, Union[str, FileDetail]]] = None):
+    if files is not None:
+        for other_filename, content in files.items():
+            with open(other_filename, mode="w", encoding="utf-8") as fh:
+                if isinstance(content, FileDetail):
+                    fh.writelines(content.contents)
+
+                    if not content.is_executable:
+                        os.chmod(other_filename, stat.S_IREAD)
+                else:
+                    fh.write(content)
+
+
+def find_and_assert_errors_matching_filename(
+    errors: List[ErrorInfo], filename: Optional[str]
+):
+    matching_errors = (
+        [err for err in errors if filename in err.filename]
+        if filename is not None
+        else errors
+    )
+
+    assert len(matching_errors) > 0, (
+        f"Expected minimum 1 error matching filename" f" {filename}, got 0"
+    )
+
+    return matching_errors
+
+
+def find_and_assert_errors_matching_location(
+    errors: List[ErrorInfo],
+    line: Optional[int] = None,
+    column: Optional[int] = None,
+    end_column: Optional[int] = None,
+):
+    def equals_or_expected_any(actual, expected):
+        return True if expected is None else actual == expected
+
+    matching_errors = [
+        x
+        for x in errors
+        if equals_or_expected_any(x.line, line)
+        and equals_or_expected_any(x.column, column)
+        and equals_or_expected_any(x.end_column, end_column)
+    ]
+
+    def none_to_star(val: Optional[int] = None):
+        return "*" if val is None else val
+
+    assert len(matching_errors) > 0, (
+        "Expected to find at least 1 error matching location"
+        f"(line={none_to_star(line)},"
+        f"column={none_to_star(column)},"
+        f"end_column{none_to_star(end_column)})"
+    )
+
+    return matching_errors
+
+
+def find_and_assert_errors_matching_message(
+    errors: List[ErrorInfo], match: Optional[str] = None
+):
+    if match is None:
+        return errors
+
+    re_match = re.compile(match)
+    matching_errors = [err for err in errors if re.search(re_match, err.message)]
+
+    assert len(matching_errors) > 0, f"Expected to find error matching message {match}"
+
+    return matching_errors
+
+
+def assert_that_config_leads_to_error(
+    config_file_contents: str,
+    expected_error: ExpectedErrorInfo,
+    config_filename: str = "test.ert",
+):
+    write_files(
+        {config_filename: config_file_contents, **(expected_error.other_files or {})}
+    )
+
+    try:
+        ErtConfig.from_file(expected_error.filename, use_new_parser=True)
+        assert False, "Parsing did not result in any errors"
```

Comment:
This is probably more cleanly handled with `with pytest.raises(ConfigValidationError) as catched_error:`

Reply:
done